src/e2e/sdk_remote_integration.spec.ts

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: Apache-2.0
+"use strict";
+
+import { TxReceipt } from "#erdstall/api/responses";
+import { Mint } from "#erdstall/api/transactions";
+import { Address } from "#erdstall/ledger";
+import { Assets } from "#erdstall/ledger/assets";
+import * as assets from "#erdstall/ledger/assets";
+import * as sdk from "#erdstall";
+import { Erdstall } from "#erdstall";
+
+import { ethers } from "ethers";
+import * as fs from "fs";
+import { exec } from "child_process";
+
+(global as any).WebSocket = require("ws");
+
+describe("Erdstall-TS-SDK", () => {
+	const opPort = 1433;
+	const opAddr = new URL(`ws://127.0.0.1:${opPort}/ws`);
+	const nodePort = 1362;
+	const mnemonic = "pistol kiwi shrug future ozone ostrich match remove crucial oblige cream critic";
+	let erdstallProcessTerminate: Promise<void>;
+
+	before(async () => {
+		await new Promise(accept => fs.writeFile("operator.cfg", JSON.stringify({
+			Mnemonic: mnemonic,
+			OperatorDerivationPath: "m/44'/60'/0'/0/0",
+			EnclaveDerivationPath:  "m/44'/60'/0'/0/1",
+			EpochDuration: 2,
+			ResponseDuration: 3,
+			PowDepth: 0,
+			RPCPort: opPort,
+			RPCHost: "0.0.0.0",
+			NodeReqTimeout: 10,
+			WaitMinedTimeout: 60
+		}), accept));
+
+		return new Promise((accept, reject) => {
+			const process = exec(["$ERDSTALL_EXE",
+				"-config operator.cfg",
+				"-epochs 10",
+				`-ledger-port ${nodePort}`
+			].join(" "), e => { if(e) reject(e); });
+
+			erdstallProcessTerminate = new Promise<void>((accept, reject) => {
+				process.on("close", accept);
+				process.on("error", reject);
+			});
+			const timeout = setTimeout(accept, 10000)
+			process.on("error", (e: Error) => {
+				clearTimeout(timeout);
+				reject(e);
+			});
+		});
```

Comment:
I hope you know what you're doing here 😂

Reply:
The setup function's promise waits for 10s or until the operator terminates in failure (or fails to execute). Additionally, another promise is created that waits for termination of the process, without time limit. The process has to be started inside the promise constructor to make the `reject` function visible to the handler I pass to the `exec` call (`on("error")` only seems to trigger if the process was also started correctly, so it cannot catch process creation failures).